dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/EditDatasetMetadataPage.java

```diff
@@ -160,6 +141,25 @@ private String returnToLatestVersion() {
         return "/dataset.xhtml?persistentId=" + dataset.getGlobalIdString() + "&version=" + workingVersion.getFriendlyVersionNumber() + "&faces-redirect=true";
     }
 
+    private void handleUpdateDatasetExceptions(Throwable throwable) {
+        if (throwable instanceof EJBException) {
+            throwable = throwable.getCause();
```

Comment:
But then it will make code more complicated, I would have to check if new throwable is null or not, or change the way code is positioned. I don't see much value in the change.

Reply:
I still consider it a bad practice but ok.

Reply:
Ok but why is that?

Reply:
Since arguments are passed by value, it may be error prone where you might expect to change value on actual parameter passed and not on its local copy created by java. Not a big dead but IMO it reduces code clearness.

Reply:
- Well but mutating the state of provided argument is anti-pattern by itself, you should not mutate state on the given parameter but make the function thread-safe. It is called pure-functions in functional programming.

Reply:
My point exactly. Code that looks like `function(parameter) { parameter = variable; }` looks like it mutates the argument, but in fact id doesn't. And it's Java specific that arguments are passed by value and that knowledge is needed to understand what and why the method does.

Reply:
Yes you are correct, but with that in mind it is hard to draw the line. Should we maybe forbid using streams from java 8 since it requires new specific knowledge? Or forbid any part of language? It is impossible, and the only way is to gain the knowledge by asking others or reading on the internet.